File: backend/src/main/java/org/globe42/web/incomes/IncomeSourceTypeDTO.java
package org.globe42.web.incomes;

import org.globe42.domain.IncomeSourceType;

/**
 * DTO for an income source type
 * @see IncomeSourceType
 * @author JB Nizet
 */
public final class IncomeSourceTypeDTO {
    private final Long id;
    private final String type;

    public IncomeSourceTypeDTO(IncomeSourceType incomeSourceType) {
        this.id = incomeSourceType.getId();
        this.type = incomeSourceType.getType();
    }

    public Long getId() {
        return id;
    }

    public String getType() {
        return type;
    }
}

File: backend/src/main/kotlin/org/globe42/web/incomes/IncomeSourceTypeDTO.kt
package org.globe42.web.incomes

import org.globe42.domain.IncomeSourceType

/**
 * DTO for an income source type
 * @see IncomeSourceType
 *
 * @author JB Nizet
 */
data class IncomeSourceTypeDTO(
        val id: Long,
        val type: String) {

    constructor(incomeSourceType: IncomeSourceType): this(
            incomeSourceType.id!!,
            incomeSourceType.type!!)
}
